src/theory/arith/bound_inference.h

```diff
@@ -27,65 +27,73 @@ namespace arith {
 
   struct Bounds
   {
-    /** The lower bound */
-    Node lower;
+    /** The lower bound value */
+    Node lower_value;
     /** Whether the lower bound is strict or weak */
     bool lower_strict = true;
+    /** The lower bound as constraint */
+    Node lower_bound;
     /** The origin of the lower bound */
     Node lower_origin;
-    /** The upper bound */
-    Node upper;
+    /** The upper bound value */
+    Node upper_value;
     /** Whether the upper bound is strict or weak */
     bool upper_strict = true;
+    /** The upper bound as constraint */
+    Node upper_bound;
     /** The origin of the upper bound */
     Node upper_origin;
   };
 
-/** Print the current variable bounds. */
-std::ostream& operator<<(std::ostream& os, const Bounds& b);
-
-/**
- * A utility class that extracts direct bounds on single variables from theory
- * atoms.
- */
-class BoundInference
-{
-  /** The currently strictest bounds for every variable. */
-  std::map<Node, Bounds> d_bounds;
-
-  /** Updates the lower bound for the given variable */
-  void update_lower_bound(const Node& origin,
-                          const Node& variable,
-                          const Node& value,
-                          bool strict);
-  /** Updates the upper bound for the given variable */
-  void update_upper_bound(const Node& origin,
-                          const Node& variable,
-                          const Node& value,
-                          bool strict);
-
- public:
-  void reset();
-
-  /**
-   * Get the current interval for v. Creates a new (full) interval if
-   * necessary.
-   */
-  Bounds& get_or_add(const Node& v);
-  /**
-   * Get the current interval for v. Returns a full interval if no interval was
-   * derived yet.
-   */
-  Bounds get(const Node& v) const;
-
-  /** Return the current variable bounds as an interval assignment. */
-  const std::map<Node, Bounds>& get() const;
+  /** Print the current bounds. */
+  std::ostream& operator<<(std::ostream& os, const Bounds& b);
 
   /**
-   * Add a new theory atom. Return true if the theory atom induces a new
-   * variable bound.
+   * A utility class that extracts direct bounds on arithmetic terms from theory
+   * atoms.
    */
-  bool add(const Node& n);
+  class BoundInference
+  {
+    /** The currently strictest bounds for every lhs. */
```

Comment:
I missed this previously, but these 3 things should go in a private block at the bottom of the class.